Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -1012,11 +1015,12 @@ void RenderObject::repaintUsingContainer(SingleThreadWeakPtr<const RenderLayerMo
         repaintContainer = &view();
 
     if (CheckedPtr fragmentedFlow = dynamicDowncast<RenderFragmentedFlow>(*repaintContainer)) {
-        fragmentedFlow->repaintRectangleInFragments(r);
+        if (fragmentedFlow)
+            fragmentedFlow->repaintRectangleInFragments(r);
         return;
     }
 
-    if (!repaintContainer)
+    if (!repaintContainer || r.isEmpty())
```

Comment:
Is this really required to fix the crash?

Reply:
This is unrelated to this crash, but might cause crashes in future so I added a check. I can remove it if you think it not needed for this radar.

Reply:
I think we should revert this change unless it's needed to fix the test case.

Reply:
ok sure